### PR TITLE
docs/dasharo-tools-suite: Split DTS documentation

### PR DESCRIPTION
**Pull request by EduKav1813**
Split the content of documentation.md file into the following structure:

documentation/
  building.md
  features.md
  running.md
  supported-hardware.md

### CODE REVIEW

File: docs/dasharo-tools-suite/documentation/building.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+## Building
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/building.md:1]:
```suggestion
# Building
```

Changes:
image
into:
image

File: docs/dasharo-tools-suite/documentation/building.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+We choose [Yocto Project](https://www.yoctoproject.org/) to prepare Dasharo
+Tools Suite system. DTS image can be built using publicly available sources.
+Thanks to publishing the build cache on
+[cache.dasharo.com](https://cache.dasharo.com/yocto/dts/) the time needed to
+finish the process should be significantly decreased.
+
+### Prerequisites
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/building.md:9]:
```suggestion
## Prerequisites
```

File: docs/dasharo-tools-suite/documentation/building.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+```bash
+mkdir yocto && cd yocto
+```
+
+```bash
+git clone https://github.com/Dasharo/meta-dts.git
+```
+
+### Build
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/building.md:36]:
```suggestion
## Build
```

File: docs/dasharo-tools-suite/documentation/building.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+local_conf_header:
+  yocto-cache: |
+    SSTATE_MIRRORS ?= "file://.* http://${LOCAL_PREMIRROR_SERVER}/${PROJECT_NAME}/sstate-cache/PATH"
+    SOURCE_MIRROR_URL ?= "http://${LOCAL_PREMIRROR_SERVER}/${PROJECT_NAME}/downloads"
+    INHERIT += "own-mirrors"
+    LOCAL_PREMIRROR_SERVER ?= "cache.dasharo.com"
+    PROJECT_NAME ?= "yocto/dts"
+```
+
+#### Build image with UEFI Secure Boot support
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/building.md:77]:
```suggestion
### Build image with UEFI Secure Boot support
```

File: docs/dasharo-tools-suite/documentation/building.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+[meta-secure-core](https://github.com/jiazhang0/meta-secure-core/). Building the
+image allow to prepare a PoC version with [uses sample
+keys](https://github.com/jiazhang0/meta-secure-core/tree/master/meta-efi-secure-boot#sample-keys)
+which by no mean should used in production. For user keys the script
+[create-user-key-store.sh](https://github.com/jiazhang0/meta-secure-core/blob/master/meta-signing-key/scripts/create-user-key-store.sh)
+can be used but it was not tested yet. Quick start with instructions on how to
+use image are described in
+[meta-efi-secure-boot](https://github.com/jiazhang0/meta-secure-core/tree/master/meta-efi-secure-boot#quick-start-for-the-first-boot).
+
+### Flash
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/building.md:107]:
```suggestion
## Flash
```

File: docs/dasharo-tools-suite/documentation/features.md [hunk 6]
```diff
@@ -1,265 +1,8 @@
...
-cd build/tmp/deploy/images/genericx86-64
-```
-
-Here the file `dts-base-image-genericx86-64.wic.gz` should be available, which
-is the image of DTS. To flash image, you can use the same command shown in
-[running section](#launching-dts_1). Just change the file name.
-
-* Boot the platform.
-
 ## Features
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/features.md:1]: Remove one `#` from all headers as in previous suggestion

**Review by m-iwanicki — CHANGES_REQUESTED** [review]
* Next time fix all pre-commit errors before asking for review as currently I stopped review as I'm just reiterating errors that pre-commit detected.
* You should also build website locally (`mkdocs serve`) and check if everything looks ok after your changes
* You should read all the warnings that are displayed by `mkserve` especially if they are different before/after your changes e.g. there is a lot of warning of type:

    ```
    WARNING -  Doc file 'unified/msi/hcl.md' contains a relative link '../../dasharo-tools-suite/documentation.md#hcl-report', but the target
               'dasharo-tools-suite/documentation.md' is not found among documentation files.
    ```

**Comment by EduKav1813** [comment]
I forget about installing pre-commit all the time 😵

File: docs/dasharo-tools-suite/documentation/building.md (hunk 1 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/building.md]: Fixed: 14a902d5faeb81ab958ef464210a7808e86a7966

File: docs/dasharo-tools-suite/documentation/building.md (hunk 2 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/building.md]: Fixed: 14a902d5faeb81ab958ef464210a7808e86a7966

File: docs/dasharo-tools-suite/documentation/building.md (hunk 3 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/building.md]: Fixed: 14a902d5faeb81ab958ef464210a7808e86a7966

File: docs/dasharo-tools-suite/documentation/building.md (hunk 4 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/building.md]: Fixed: 14a902d5faeb81ab958ef464210a7808e86a7966

File: docs/dasharo-tools-suite/documentation/building.md (hunk 5 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/building.md]: Fixed: 14a902d5faeb81ab958ef464210a7808e86a7966

File: docs/dasharo-tools-suite/documentation/features.md (hunk 6 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md]: Fixed: 14a902d5faeb81ab958ef464210a7808e86a7966

**Comment by EduKav1813** [comment]
@m-iwanicki Fixed headers, pre-commit and mkdocs issues. All checks pass. Also merged master branch for compatibility.

File: docs/dasharo-tools-suite/documentation/running.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+### Requirements
+
+Below are the requirements that must be met to run DTS from a USB device on the
+platform:
+
+* USB stick (at least 2GB),
+* Latest image from [releases](https://github.com/Dasharo/meta-dts/releases)
+  section.
+  with Dasharo).
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/running.md:65]:
```suggestion
```

I think this is copy/paste typo.

File: docs/dasharo-tools-suite/documentation/features.md [hunk 8]
```diff
@@ -270,19 +13,42 @@ This section describes the functionality of the Dasharo Tools Suite. These are:
...
 
-### Dasharo zero-touch initial deployment
+## Available Commands
+
+When DTS is started, it has following options for the user to choose from:
+
+* **(1)** Dasharo HCL Report:
+
+  Generate HCL report on the current machine.
+  Read more: [HCLReport](#hcl-report)
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/features.md:23]:
```suggestion
    Generate HCL report on the current machine.
    Read more: [HCLReport](#hcl-report)
```

File: docs/dasharo-tools-suite/documentation/features.md [hunk 9]
```diff
@@ -270,19 +13,42 @@ This section describes the functionality of the Dasharo Tools Suite. These are:
     + [run commands from iPXE shell automatically](#run-commands-from-ipxe-shell-automatically),
     + [run DTS using VentoyOS](#run-dts-using-ventoyos).
 
-### Dasharo zero-touch initial deployment
+## Available Commands
+
+When DTS is started, it has following options for the user to choose from:
+
+* **(1)** Dasharo HCL Report:
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/features.md:20]: Maybe remove `(` (here and below) so it'll be same as in DTS?

File: docs/dasharo-tools-suite/documentation/features.md [hunk 10]
```diff
@@ -270,19 +13,42 @@ This section describes the functionality of the Dasharo Tools Suite. These are:
...
+
+  Generate HCL report on the current machine.
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/features.md:22]: You could just use:
* **1)** [Dasharo HCL Report](#hcl-report) - generate Hardware Compatibility List Report

File: docs/dasharo-tools-suite/documentation/features.md [hunk 11]
```diff
@@ -270,19 +13,42 @@ This section describes the functionality of the Dasharo Tools Suite. These are:
...
+
+Load your Dasharo Pro Package (DPP) keys. Read more:
+[How can I use my Dasharo Pro Package credentials](../../osf-trivia-list/dts.md#how-can-i-use-my-dasharo-pro-package-credentials)
+
+* **(R)** Reboot
+* **(P)** Poweroff
+* **(S)** Enter shell
+* **(K)** Launch SSH Server
+* **(L)** Enable sending DTS logs
+* **(V)** Enable verbose mode
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/features.md:37]:
```suggestion* **1)** [Dasharo HCL Report](#hcl-report)
* **2)** [Update Dasharo Firmware](#firmware-update) or [Install Dasharo Firmware](#dasharo-zero-touch-initial-deployment)
* **3)** [Restore Firmware from Dasharo HCL Report](#update-issues)
* **4)** [Load your DPP keys](../../osf-trivia-list/dts.md#how-can-i-use-my-dasharo-pro-package-credentials)
    \- Load your Dasharo Pro Package (DPP) keys
* **R** Reboot
* **P** Poweroff
* **S** Enter shell
* **K** Launch SSH Server
* **L** [Enable sending DTS logs](../../osf-trivia-list/dts.md#how-can-i-help-the-support-team-diagnose-my-problem-faster)
* **V** [Enable verbose mode](../../osf-trivia-list/dts.md#how-can-i-help-the-support-team-diagnose-my-problem-faster)
```

Maybe like this?

Comment (m-iwanicki) [reply docs/dasharo-tools-suite/documentation/features.md:37]: @DaniilKl there is also [5) DTS extensions](https://github.com/Dasharo/dts-scripts/blob/c7b8d036f549c41f848ee254ba309baf762835d5/include/dts-functions.sh#L1346), is it already available and if yes can you help to expand this docs or point to documentation?

File: docs/dasharo-tools-suite/documentation/features.md [hunk 12]
```diff
@@ -561,24 +327,24 @@ If you used `flashrom` as described in [Local firmware
...
+* [NovaCustom V540TNx](../../variants/novacustom_v540tnx/releases.md),
+* [NovaCustom V560TU](../../variants/novacustom_v560tu/releases.md)
+* [NovaCustom V560TNx](../../variants/novacustom_v560tnx/releases.md).
 
 Starting from DTS v1.2.0 to perform EC transition please run
 [firmware update](#firmware-update) on the platform with proprietary vendor EC
 firmware.
 
-### EC update
+## EC update
```
Comment (m-iwanicki) [inline docs/dasharo-tools-suite/documentation/features.md:347]: Could you fix indent:

````diff
* Retrieve information about your current EC.

-  ```bash
-  dasharo_ectool info
-  ```

-  The output of the above-described command should contain information about
-  the version of flashed firmware:

-  ```bash
-  board: clevo/ns50mu
-  version: 2022-08-16_c12ff1a
-  ```
+    ```bash
+    dasharo_ectool info
+    ```

+    The output of the above-described command should contain information about
+    the version of flashed firmware:

+    ```bash
+    board: clevo/ns50mu
+    version: 2022-08-16_c12ff1a
+    ```

* Download the newest version of Embedded Controller firmware.
* Plug in power supply, without it, flashing EC is not possible as losing power
  may cause in damaged firmware.
* Flash Embedded Controller firmware internally.

-  ```bash
-  dasharo_ectool flash ec_file.rom
-  ```

-  The output of the above-described command should look as follows:

-  ```bash
-  file board: Ok("clevo/ns50mu")
-  file version: Ok("2022-08-16_c12ff1a")
-  ec board: Ok("clevo/ns50mu")
-  ec version: Ok("2022-08-31_cbff21b")
-  Waiting 5 seconds for all keys to be released
-  Sync
-  SPI Read 128K
-  Saving ROM to backup.rom
-  SPI Write 128K
-  SPI Read 128K
-  Successfully programmed SPI ROM
-  Result: Ok(())
-  Sync
-  System will shut off in 5 seconds
-  Sync
-  ```

-  > Note: this is example output, versions may differ

+    ```bash
+    dasharo_ectool flash ec_file.rom
+    ```

+    The output of the above-described command should look as follows:

+    ```bash
+    file board: Ok("clevo/ns50mu")
+    file version: Ok("2022-08-16_c12ff1a")
+    ec board: Ok("clevo/ns50mu")
+    ec version: Ok("2022-08-31_cbff21b")
+    Waiting 5 seconds for all keys to be released
+    Sync
+    SPI Read 128K
+    Saving ROM to backup.rom
+    SPI Write 128K
+    SPI Read 128K
+    Successfully programmed SPI ROM
+    Result: Ok(())
+    Sync
+    System will shut off in 5 seconds
+    Sync
+    ```

+    > Note: this is example output, versions may differ

* Computer will shut down automatically.
* Power on your computer. Booting process may take a while.
* After boot, choose option `S` to drop to Shell.
* Retrieve information about your updated EC.

-  ```bash
-  dasharo_ectool info
-  ```

-  The output of the above-described command should contain information about
-  the version of flashed firmware:

-  ```bash
-  board: clevo/ns50mu
-  version: 2022-08-31_cbff21b
-  ```
+    ```bash
+    dasharo_ectool info
+    ```

+    The output of the above-described command should contain information about
+    the version of flashed firmware:

+    ```bash
+    board: clevo/ns50mu
+    version: 2022-08-31_cbff21b
+    ```
````

**Review by m-iwanicki — CHANGES_REQUESTED** [review]

File: docs/dasharo-tools-suite/documentation/features.md (hunk 8 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md]: Also fixed by b133540c400db01df4c8bcaf581f2103c23cd870

File: docs/dasharo-tools-suite/documentation/features.md (hunk 9 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md]: Outdated due to: b133540c400db01df4c8bcaf581f2103c23cd870

File: docs/dasharo-tools-suite/documentation/features.md (hunk 10 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md]: Covered by: b133540c400db01df4c8bcaf581f2103c23cd870

File: docs/dasharo-tools-suite/documentation/features.md (hunk 11 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md]: Introduced this style in: b133540c400db01df4c8bcaf581f2103c23cd870

File: docs/dasharo-tools-suite/documentation/features.md (hunk 12 above)
Comment (m-iwanicki) [reply docs/dasharo-tools-suite/documentation/features.md:345]: If you looked at the diff I posted you would see I added 2 spaces to make it proper indent:

Without indent:

image

With indent:

image

Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md:345]: Ah, I build it and checked once again and now I see. Mkdocs wants one more level of indent for it to be displayed properly.

Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md:345]: 90f9fd4c503a04a984a9016614ed7cef2c6c505a

Comment (m-iwanicki) [reply docs/dasharo-tools-suite/documentation/features.md:345]: It probably depends but I saw that a lot of times markdown viewers don't accept 2 spaces as indent by default (unless configured somehow).

But printscreen wasn't made on GitHub but on local website run by using `mkdocs serve`

**Review by m-iwanicki — APPROVED** [review]

File: docs/dasharo-tools-suite/documentation/supported-hardware.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
+# Supported hardware
+
+Dasharo Tools Suite was prepared to run on x86 platforms, but we can confirm
+that it boots on the following platforms:
+
+* ASUS KGPE-D16,
```
Comment (artur-rs) [inline docs/dasharo-tools-suite/documentation/supported-hardware.md:6]: @EduKav1813 @macpijan if I remember correctly, we should drop this from the list and DTS logic

File: docs/dasharo-tools-suite/documentation/features.md [hunk 14]
```diff
@@ -313,17 +77,17 @@ And partially (only EC firmware flashing) on:
 * NovaCustom V540TU/TNx,
```
Comment (artur-rs) [inline docs/dasharo-tools-suite/documentation/features.md:77]: @EduKav1813 let's move section related to HW `This feature is supported on the following platforms:` to the `supported-hardware.md` file. Then, let's link to the corresponding file.

**Review by artur-rs — CHANGES_REQUESTED** [review]

File: docs/dasharo-tools-suite/documentation/features.md (hunk 14 above)
Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/features.md:77]: Resolved here: 499b1cf0d4e7b3b7150e604595cac99bd60d4000

File: docs/dasharo-tools-suite/documentation/supported-hardware.md (hunk 13 above)
Comment (m-iwanicki) [reply docs/dasharo-tools-suite/documentation/supported-hardware.md]: https://github.com/Dasharo/dts-scripts/pull/50 - PR removing ASUS support in DTS

Comment (EduKav1813) [reply docs/dasharo-tools-suite/documentation/supported-hardware.md]: I am not sure, but here is the commit that removes this hardware from the list: db9a619e36317c3f4930693677637f379eb21105

We can revert it if not needed

**Review by artur-rs — APPROVED** [review]

**Pull request merged**